python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -259,21 +332,331 @@ def scaleModel(self, model, star, inPlace=True, nb90Rots=0):
+        return annulusStat.getValue()
+
+    def findPsfAnnularFlux(self, brightStarStamp, maskedModel):
+        """Find the annular flux of the PSF model at the radius where the flux of the given star is
+        determined.
+
+        Parameters
+        ----------
+        brightStarStamp : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            A stamp of a bright star to be subtracted.
+        maskedModel : `~lsst.afw.image.MaskedImageF`
+            A masked image of the PSF model.
+
+        Returns
+        -------
+        annularFlux: float
+            The annular flux of the PSF model at the radius where the flux of the given star is determined.
+        """
+        annulusImage = self.initAnnulusImage()
+        annulus = self.createAnnulus(brightStarStamp)
+        annulus.copyMaskedImage(maskedModel, annulusImage)
+        annularFlux = self.applyStatsControl(annulusImage)
+        return annularFlux
+
+    def findPsfAnnularFluxes(self, brightStarStamps):
+        """Find the annular fluxes of the given PSF model.
+
+        Parameters
+        ----------
+        brightStarStamps : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            The stamps of stars that will be subtracted from the exposure.
+
+        Returns
+        -------
+        PsfAnnularFluxes: numpy.array
+            A two column numpy.array containing annular fluxes of the PSF at radii where the flux for stars
+            exist (could be found).
+
+        Notes
+        -----
+        While the PSF model is normalized at a certain radius, the flux of a star at that radius might be
+        impossible to find. Therefore, we have to scale the PSF model considering a radius where the star has
+        an identified flux. To do that, the flux of the model should be found and used to adjust the scaling
+        step.
+        """
+        outerRadii = []
+        annularFluxes = []
+        maskedModel = MaskedImageF(self.model.image)
+        # the model has wrong bbox values. Should be fixed in extended_psf.py?
+        maskedModel.setXY0(0, 0)
+        for star in brightStarStamps:
+            if star.optimalOuterRadius not in outerRadii:
+                annularFlux = self.findPsfAnnularFlux(star, maskedModel)
+                outerRadii.append(star.optimalOuterRadius)
+                annularFluxes.append(annularFlux)
+        return np.array([outerRadii, annularFluxes]).T
+
+    def preparePlaneModelStamp(self, brightStarStamp):
+        """Prepare the PSF model before scaling.
+
+        Parameters
+        ----------
+        brightStarStamp : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            The stamp of the star to which the PSF model will be scaled.
+
+        Returns
+        -------
+        bbox: `~lsst.geom.Box2I`
+            Contains the corner coordination and the dimensions of the model stamp.
+
+        invImage: `~lsst.afw.image.MaskedImageF`
```

Comment:
A note is added to the docstring to make that clear.

Basically, since detectors have different orientations, the PSF model should be rotated to match the orientation of the detectors in some cases. To do that, the code uses the inverse of the transform that is applied to the bright star stamp to match the orientation of the detector.